### PR TITLE
feat(cli): update TypeScript template to match aws-cdk layout

### LINKED ISSUE #3006: cli: match cdk file structure
### Description

As an experienced CDK (CloudFormation) user, I was surprised that the layout of the CDKTF application was so "flat":

```
# CDKTF
> tree . -L 2 -I 'node_modules'
.
├── __tests__
│   └── main-test.ts
├── cdktf.json
├── help
├── jest.config.js
├── main.ts
├── package-lock.json
├── package.json
├── setup.js
└── tsconfig.json
```

```
# CDK
tree . -L 2 -I 'node_modules'
.
├── README.md
├── bin
│   └── test-cdk.ts
├── cdk.json
├── jest.config.js
├── lib
│   └── test-cdk-stack.ts
├── package-lock.json
├── package.json
├── test
│   └── test-cdk.test.ts
└── tsconfig.json
```

I think it'd be nice to emulate the exact same layout, as I believe there will be a good amount of crossover between CloudFormation CDK and CDKTF. Specifically:

- [ ] move `new App()` to `bin/{{app-name}}.ts`
- [ ] move initial stack to `lib/{{app-name}}-stack.ts`
- [ ] rename `__tests__` to `test`

### References

https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk/lib/init-templates/app/typescript

### Help Wanted

- [X] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by blimmer**
### Related issue

Fixes #3006

### Description

This PR adjusts the output format for CDKTF TypeScript projects created by `cdktf init --template typescript` to better match CDK. This makes it simpler for new CDKTF users with CDK experience to understand where the files are located.

You can see the current CDK layout here: https://github.com/aws/aws-cdk/tree/06bea3159e137a8e85e362795ef2cf7e219e381c/packages/aws-cdk/lib/init-templates/app/typescript

or, generate a new CDK project for comparison

```shell
npx aws-cdk@latest init app --language=typescript
```

### Checklist

- [x] I have updated the PR title to match [CDKTF's style guide](https://github.com/hashicorp/terraform-cdk/blob/main/CONTRIBUTING.md#pull-requests-1)
- [x] I have run the linter on my code locally
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if applicable
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works if applicable
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/@cdktf/cli-core/templates/typescript/lib/{{$base}}-stack.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Construct } from "constructs";
+import { TerraformStack } from "cdktf";
+
+export class {{basePascal}}Stack extends TerraformStack {
```
Comment (blimmer) [inline packages/@cdktf/cli-core/templates/typescript/lib/{{$base}}-stack.ts:4]: For an app named foo `MyStack` becomes `FooStack`, just like CDK.

File: packages/@cdktf/cli-core/templates/typescript/jest.config.ts
Comment (blimmer) [inline packages/@cdktf/cli-core/templates/typescript/jest.config.ts:1]: Updating this file to use a TypeScript config vs. JS. I retained all the existing values.

File: packages/@cdktf/cli-core/templates/typescript/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+/*
+ * For a detailed explanation regarding each configuration property, visit:
+ * https://jestjs.io/docs/configuration
+ */
+const config: Config = {
+  preset: "ts-jest",
+  clearMocks: true,
+  coverageProvider: "v8",
+  moduleFileExtensions: ["ts", "js", "json", "node"],
+  setupFilesAfterEnv: ["<rootDir>/test/setup.js"],
```
Comment (blimmer) [inline packages/@cdktf/cli-core/templates/typescript/jest.config.ts:12]: Updated from `<rootDir>/setup.js`

File: packages/@cdktf/cli-core/templates/typescript/jest.config.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  testEnvironment: "node",
+  testMatch: ["**/test/**/*.ts", "**/?(*.)+(spec|test).ts"],
```
Comment (blimmer) [inline packages/@cdktf/cli-core/templates/typescript/jest.config.ts:14]: Updated from `**/__tests__/**/*.ts`

File: packages/@cdktf/cli-core/templates/typescript/bin/{{$base}}.ts
Comment (blimmer) [inline packages/@cdktf/cli-core/templates/typescript/bin/{{$base}}.ts:1]: Now, just the app is created here. The stack is created in a separate file, as with CDK.

File: packages/@cdktf/cli-core/.eslintignore [hunk 4]
```diff
@@ -1,2 +1,5 @@
 # ignore build output
-*.d.ts
\ No newline at end of file
+*.d.ts
+
+# ignore templates, since they're not valid TypeScript
+templates
```
Comment (blimmer) [inline packages/@cdktf/cli-core/.eslintignore:5]: Now that we're using syntax like:

```
export class {{basePascal}}Stack extends TerraformStack {
```

We have invalid TS, so eslint blows up.

**Comment by blimmer** [comment]
Ah, I need to update the website docs. Taking back to draft.

**Comment by blimmer** [comment]
I think I'll need to update at least:

- [ ] Unit tests
- [ ] Docs
- [ ] TypeScript examples

to complete this PR. Since that'd be a decent time commitment, @DanielMSchmidt, is this a PR the team would consider merging? If not, I'll save the time and close it. If so, I'm happy to complete it.

**Comment by xiehan** [comment]
Hi @blimmer:

Thank you for the effort you put into this PR! We discussed it as a team and we agreed that while there are a couple of valid arguments for making this change, overall, this is a bigger refactor than we're willing to take on; if we were going to proceed with this, we would also want to make sure that all of our tutorials and example projects outside of this repo get updated to the new format as well so that we're being consistent across-the-board. That additional effort outweighs the benefits of making this change from our perspective.

Again, we do acknowledge your contribution and appreciate the time you spent on this!

**Pull request closed without merging**